File: src/2389. Longest Subsequence With Limited Sum.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @return {number[]}
 */

export function answerQueries(nums, queries) {
  nums.sort((a, b) => a - b)
  const results = []
  for (let query of queries) {
    let i = 0
    while (i < nums.length && query >= nums[i]) {
      query -= nums[i]
      i++
    }
    results.push(i)
  }
  return results
}

File: src/2389. Longest Subsequence With Limited Sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
        sort(nums.begin(), nums.end());
        vector<int> results;
        for (auto &query : queries) {
            int i = 0;
            while (i < nums.size() && query >= nums[i]) {
                query -= nums[i];
                i++;
            }
            results.push_back(i);
        }
        return results;
    }
};